### PR TITLE
New timeline FFI API

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

**Review by gnunicorn — APPROVED** [review]
lftm

File: bindings/matrix-sdk-ffi/Cargo.toml [hunk 1]
```diff
@@ -12,14 +12,18 @@ repository = "https://github.com/matrix-org/matrix-rust-sdk"
...
+
 [build-dependencies]
 uniffi_build = { git = "https://github.com/mozilla/uniffi-rs", rev = "0eee77f67b716c4896494606e5931d249871b23a", features = ["builtin-bindgen"] }
 
 [dependencies]
 anyhow = "1.0.51"
 extension-trait = "1.0.1"
 futures-core = "0.3.17"
-futures-signals = { version = "0.3.28" }
+futures-signals = { version = "0.3.30", default-features = false }
```
[the remark this answers is not in the document]
Comment (jplatte) [reply bindings/matrix-sdk-ffi/Cargo.toml:26]: `debug` and `serde` features. Not sure what exactly the first one is about, but things still compile with both off.

File: bindings/matrix-sdk-ffi/src/api.udl [hunk 2]
```diff
@@ -257,50 +261,141 @@ interface Room {
...
 };
 
-interface BackwardsStream {
-    sequence<AnyMessage> paginate_backwards(u64 count);
+callback interface TimelineListener {
+    void on_update(TimelineDiff update);
 };
 
-interface RoomMessageEventContent {};
+// Should really be an enum
```
[the remark this answers is not in the document]
Comment (jplatte) [reply bindings/matrix-sdk-ffi/src/api.udl:268]: Why?

[the remark this answers is not in the document]
Comment (gnunicorn) [reply bindings/matrix-sdk-ffi/src/api.udl:268]: What does it add? Either we should make that an issue in the issue tracker or remove it. But old authors comment on how code wasn't what they wanted when they wrote it doesn't really add value for anyone reading it later...

**Pull request merged**